components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -0,0 +1,183 @@
+            show={props.show}
+            role='dialog'
+            aria-labelledby='editPostTimeModalLabel'
+            onHide={props.onClose}
+        >
+            <Modal.Header closeButton={true}>
+                <Modal.Title
+                    componentClass='h1'
+                    id='editPostTimeModalLabel'
+                >
+                    <FormattedMessage
+                        id='edit_post.time_limit_modal.title'
+                        defaultMessage='Configure Global Edit Post Time Limit'
+                    />
+                </Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+                <FormattedMarkdownMessage
+                    id='edit_post.time_limit_modal.description'
+                    defaultMessage='Setting a time limit **applies to all users** who have the "Edit Post" permissions in any permission scheme.'
+                />
+                <div className='pl-3 pb-3 pt-3'>
+                    <div className='pt-3'>
+                        <input
+                            id='anytime'
+                            type='radio'
+                            name='limit'
+                            value={Constants.ALLOW_EDIT_POST_ALWAYS}
+                            checked={alwaysAllowPostEditing}
+                            onChange={handleOptionChange}
+                        />
+                        <label htmlFor='anytime'>
+                            <FormattedMessage
+                                id='edit_post.time_limit_modal.option_label_anytime'
+                                defaultMessage='Anytime'
+                            />
+                        </label>
+                    </div>
+                    <div className='pt-2'>
+                        <input
+                            id='timelimit'
+                            type='radio'
+                            name='limit'
+                            value={Constants.ALLOW_EDIT_POST_TIME_LIMIT}
+                            checked={!alwaysAllowPostEditing}
+                            onChange={handleOptionChange}
+                        />
+                        <label htmlFor='timelimit'>
+                            <FormattedMessage
+                                id='edit_post.time_limit_modal.option_label_time_limit.preinput'
+                                defaultMessage='Can edit for'
+                            />
+                        </label>
+                        <input
+                            type='number'
+                            className='form-control inline'
+                            min='0'
+                            step='1'
+                            max={INT32_MAX}
+                            id='editPostTimeLimit'
+                            readOnly={alwaysAllowPostEditing}
+                            onChange={handleSecondsChange}
+                            value={alwaysAllowPostEditing ? '' : postEditTimeLimit}
```

Comment:
regarding the comment from above: When the value for `alwaysAllowPostEditing` is `true` it will show nothing in the input, but once it is set to `false` the stored value will get inserted here